wicker/core/datasets.py

```diff
@@ -22,6 +24,154 @@
 FILE_LOCK_TIMEOUT_SECONDS = 300
 
 
+def get_file_size_s3_proced(buckets_keys: List[Tuple[int, int]]) -> int:
+    """Get file size of s3 files, most often column files.
+
+    This works on any list of buckets and keys but is generally only
+    used for column files as those are the majority of what is stored on
+    s3 for Wicker. Wicker also stores parquet files on s3 but those are limited
+    to one file per dataset and one schema file.
+
+    This splits your buckets_keys_list across mutliple processes on your local host
+    where each process is then multi threaded further. This reduces the i/o wait to
+    what is assumed to be the absolute minimum that is possible.
+
+    Args:
+        buckets_keys: (List[Tuple[str, str]]): A list of buckets and keys for which
+        to fetch size in bytes on s3. Tuple index 0 is bucket and index 1 is key of the file.
+
+    Returns:
+        int size of file list in bytes.
+    """
+    # declare the variables neccessary to split
+    # out between multiple procs
+    manager = Manager()
+    total_size = manager.Value("i", 0)
+    lock = manager.Lock()
+    buckets_keys_list = list(buckets_keys)
+
+    buckets_keys_chunks = chunk_data_for_split(
+        chunkable_data=buckets_keys_list, chunk_additions=[total_size, lock], chunk_number=200
+    )
+
+    print("Grabbing file information from s3 heads")
+    # set the proc pool up and distribute the data between
+    # along with shared values
+    # reserve one cpu for other operations
+    pool = Pool(cpu_count() - 1)
+    # arbitrary chunking breaks typing here, still keep typing on function
+    # for documentation
+    pool.map(get_file_size_s3_threaded, buckets_keys_chunks)  # type: ignore
```

Comment:
@pickles-bread-and-butter so if you look carefully, you actually do not need the `lock` or `total_size` shared memory value (which will enable you to greatly simplify your logic here). If you look at https://docs.python.org/3/library/multiprocessing.html#introduction you can see that `pool.map` will return a list of the values returned from the mapped function (i.e. `get_file_size_s3_threaded`). So then what you need to do is change `get_file_size_s3_threaded` so that it returns an `int` also (so it should return the computed size of the input chunk that was assigned).

Reply:
Then of course you sum up all the entries in the list returned from `pool.map`.

Reply:
I think the sum on that would be less efficient than doing the values iteratively. ie: you have to process it once to get the byte values and then pass again to get the sum so it would be 2O(N) (sum of list) v O(N) (shared memory). That was the logic here, let me know if you still want it changed.

Reply:
Asked offline, went with the suggested sum